pkg/sql/opt/optbuilder/insert.go

```diff
@@ -371,8 +371,8 @@ func (mb *mutationBuilder) needExistingRows() bool {
 			// #1: Don't consider key columns.
 			continue
 		}
-		if cat.IsSystemColumn(mb.tab, i) {
-			// #2: Don't consider system columns.
+		if kind := mb.tab.ColumnKind(i); kind == cat.System || kind == cat.Virtual {
```

Comment:
Maybe search for the wanted kinds instead of unwanted kinds so that if we add a new kind in the future this won't break.